### PR TITLE
fix: avoid allocation too much in start up packets

### LINKED ISSUE #7345: panicked at `capacity overflow`, library/alloc/src/raw_vec.rs
### Describe the bug

version = 0.1.15

```
thread ‘risingwave-main’ panicked at ‘capacity overflow’, library/alloc/src/raw_vec.rs:518:5
stack backtrace:
   0: rust_begin_unwind
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/core/src/panicking.rs:64:14
   2: alloc::raw_vec::capacity_overflow
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/alloc/src/raw_vec.rs:518:5
   3: alloc::raw_vec::RawVec<T,A>::allocate_in
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/alloc/src/raw_vec.rs:178:27
   4: alloc::raw_vec::RawVec<T,A>::with_capacity_zeroed_in
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/alloc/src/raw_vec.rs:139:9
   5: <u8 as alloc::vec::spec_from_elem::SpecFromElem>::from_elem
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/alloc/src/vec/spec_from_elem.rs:52:31
   6: alloc::vec::from_elem
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/alloc/src/vec/mod.rs:2557:5
   7: pgwire::pg_message::FeStartupMessage::read::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_message.rs:336:27
   8: pgwire::pg_protocol::PgStream<S>::read_startup::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_protocol.rs:567:46
   9: pgwire::pg_protocol::Conn<S>::read_startup::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_protocol.rs:650:53
  10: pgwire::pg_protocol::PgProtocol<S,SM,VS>::read_message::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_protocol.rs:225:67
  11: pgwire::pg_server::handle_connection::{{closure}}::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_server.rs:145:46
  12: pgwire::pg_server::handle_connection::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_server.rs:131:1
  13: pgwire::pg_server::pg_serve::{{closure}}::{{closure}}
             at ./risingwave/src/utils/pgwire/src/pg_server.rs:118:40
  14: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.37/src/instrument.rs:272:9
  15: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/core.rs:208:17
  16: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/loom/std/unsafe_cell.rs:14:9
  17: tokio::runtime::task::core::Core<T,S>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/core.rs:197:13
  18: tokio::runtime::task::harness::poll_future::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:483:19
  19: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/core/src/panic/unwind_safe.rs:271:9
  20: std::panicking::try::do_call
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panicking.rs:483:40
  21: std::panicking::try
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panicking.rs:447:19
  22: std::panic::catch_unwind
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panic.rs:137:14
  23: tokio::runtime::task::harness::poll_future
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:471:18
  24: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:107:27
  25: tokio::runtime::task::harness::Harness<T,S>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:61:15
  26: tokio::runtime::task::raw::RawTask::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/raw.rs:134:18
  27: tokio::runtime::task::LocalNotified<S>::run
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/mod.rs:430:9
  28: tokio::runtime::scheduler::multi_thread::worker::Context::run_task::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/worker.rs:439:13
  29: tokio::runtime::coop::with_budget
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/coop.rs:102:5
  30: tokio::runtime::coop::budget
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/coop.rs:68:5
  31: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/worker.rs:438:9
  32: tokio::runtime::scheduler::multi_thread::worker::Context::run
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/worker.rs:405:24
  33: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/worker.rs:390:17
  34: tokio::macros::scoped_tls::ScopedKey<T>::set
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/macros/scoped_tls.rs:61:9
  35: tokio::runtime::scheduler::multi_thread::worker::run
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/worker.rs:387:5
  36: tokio::runtime::scheduler::multi_thread::worker::Launch::launch::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/worker.rs:365:45
  37: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/blocking/task.rs:42:21
  38: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.37/src/instrument.rs:272:9
  39: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/core.rs:208:17
  40: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/loom/std/unsafe_cell.rs:14:9
  41: tokio::runtime::task::core::Core<T,S>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/core.rs:197:13
  42: tokio::runtime::task::harness::poll_future::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:483:19
  43: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/core/src/panic/unwind_safe.rs:271:9
  44: std::panicking::try::do_call
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panicking.rs:483:40
  45: std::panicking::try
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panicking.rs:447:19
  46: std::panic::catch_unwind
             at ./rustc/bdb07a8ec8e77aa10fb84fae1d4ff71c21180bb4/library/std/src/panic.rs:137:14
  47: tokio::runtime::task::harness::poll_future
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:471:18
  48: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:107:27
  49: tokio::runtime::task::harness::Harness<T,S>::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/harness.rs:61:15
  50: tokio::runtime::task::raw::RawTask::poll
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/raw.rs:134:18
  51: tokio::runtime::task::UnownedTask<S>::run
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/task/mod.rs:467:9
  52: tokio::runtime::blocking::pool::Task::run
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/blocking/pool.rs:159:9
  53: tokio::runtime::blocking::pool::Inner::run
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/blocking/pool.rs:510:17
  54: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at ./root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/blocking/pool.rs:468:13
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

From the slack community: https://risingwave-community.slack.com/archives/C02T3F7UYM6/p1673502585719499

### PR DESCRIPTION
**Pull request by BowenXiao1999**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Close #7345

We should avoid large memory allocation: https://doc.rust-lang.org/src/alloc/raw_vec.rs.html#505

This is reported by community, can not reproduce so no test

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by BugenZhao** [comment]
I'm curious how it happens. If there's no malicious requests, is it caused by some compatibility issues?

File: src/utils/pgwire/src/pg_message.rs [hunk 1]
```diff
@@ -332,8 +332,14 @@ impl FeStartupMessage {
     pub async fn read(stream: &mut (impl AsyncRead + Unpin)) -> Result<FeMessage> {
         let len = stream.read_i32().await?;
         let protocol_num = stream.read_i32().await?;
-        let payload_len = len - 8;
-        let mut payload = vec![0; payload_len as usize];
+        let payload_len = (len - 8) as usize;
+        if payload_len >= isize::MAX as usize {
```
Comment (liurenjie1024) [inline src/utils/pgwire/src/pg_message.rs:336]: I think it's better to make this a configuration rather a constant

**Comment by BowenXiao1999** [comment]
> I'm curious how it happens. If there's no malicious requests, is it caused by some compatibility issues?

Yes, I was also curious. The report context https://github.com/risingwavelabs/risingwave/issues/7345#issuecomment-1384790397.

I think it is not compatibility, as the packet must come from the psql binary client. Maybe we misundertand some packet layout, but anyway, such check looks OK to me. https://github.com/risingwavelabs/risingwave/issues/7345#issuecomment-1384790397

File: src/utils/pgwire/src/pg_message.rs (hunk 1 above)
Comment (BowenXiao1999) [reply src/utils/pgwire/src/pg_message.rs:336]: This comes from here: https://doc.rust-lang.org/src/alloc/raw_vec.rs.html#505

Actually i don't think we should make it a config, cuz even devs are not expected to roll it down or up in most cases, not talk about users. This is just avoid the capacity overflow problem.

**Review by liurenjie1024 — APPROVED** [review]
Let's fix it first

**Pull request merged**